halo2-base/src/gates/range.rs

```diff
@@ -144,17 +174,32 @@ impl<F: ScalarField> RangeConfig<F> {
     }
 }
 
+/// Trait that implements methods to constrain a field element number `x` is within a certain number of bits.
 pub trait RangeInstructions<F: ScalarField> {
+
+    ///TODO: Add Description
+
+    /// The type of Gate used within the instructions
     type Gate: GateInstructions<F>;
 
+    /// Returns the gate configuration for this range.
     fn gate(&self) -> &Self::Gate;
+
+    /// Returns the gate strategy for this range.
     fn strategy(&self) -> RangeStrategy;
 
+    /// Returns the number of bits the lookup table represents
     fn lookup_bits(&self) -> usize;
 
-    /// Constrain that `a` lies in the range [0, 2<sup>range_bits</sup>).
+    /// Checks that `a` lies in the range [0, 2<sup>range_bits</sup>).
+    /// * a: value to be range checked
+    /// * range_bits: number of bits to represent the range
     fn range_check(&self, ctx: &mut Context<F>, a: AssignedValue<F>, range_bits: usize);
 
+    /// Check that 'a' is less than 'b' 
+    /// * a: value to check
+    /// * b: upper bound
+    /// * num_bits: number of bits to represent the range
```

Comment:
Clarify that it is *assumed* that both inputs `a` and `b` are already known to have `<= num_bits`. There would be undefined behavior if this assumption is not satisfied. 

These are the type of assumptions that we need to take extra care to document for security.

Reply:
See https://github.com/axiom-crypto/halo2-lib/blob/61e516d5a9bf847428cfbc633d488bca83d86c31/halo2-base/src/gates/range.rs#L431 (should be in the trait implementation though)